packages/@okta/vuepress-theme-prose/components/SmartLink.vue

```diff
@@ -0,0 +1,100 @@
+<template>
+  <RouterLink
+    v-if="isInternalURI"
+    :class="classes"
+    :to="item.link"
+    :exact="exact"
+  >
+    <slot v-if="$slots.default"></slot>
+    <span v-else-if="item.text">{{ item.text }}</span>
+  </RouterLink>
+  <a v-else :href="item.link" :class="classes" :target="target" :rel="rel">
+    <slot v-if="$slots.default"></slot>
+    <span v-else-if="item.text">{{ item.text }}</span>
+  </a>
+</template>
+
+<script>
+const EXTERNAL_REGEX = /^[a-z]+:/;
+export function isExternal(path) {
+  return EXTERNAL_REGEX.test(path);
+}
+
+const MAILTO_REGEX = /^mailto:/;
+export function isMailto(path) {
+  return MAILTO_REGEX.test(path);
+}
+
+export default {
+  name: "SmartLink",
+  props: {
+    item: {
+      required: true,
+    },
+    classes: {
+      type: String,
+      default: "",
+    },
+  },
+  data() {
+    return {
+      isSearchPage: false,
+    };
+  },
+  beforeMount() {
+    this.setIsSearchPage();
+  },
+  watch: {
+    $route: "setIsSearchPage",
+  },
+  computed: {
+    exact() {
+      // Exact mode required to prevent links targeting the root to always be active
+      return this.item.link === "/";
+    },
+
+    isMailToURI() {
+      return isMailto(this.item.link);
+    },
+
+    isBlankTarget() {
+      return this.target === "_blank";
+    },
+
+    // Used to force page reload when link is external, blank or from the search page
+    isInternalURI() {
+      return (
+        !isExternal(this.item.link) && !this.isBlankTarget && !this.isSearchPage
+      );
+    },
+
+    target() {
+      if (this.isMailToURI) {
+        return null;
+      }
+      if (this.item.target) {
+        return this.item.target;
+      }
+      return isExternal(this.item.link) ? "_blank" : "";
+    },
+    
+    rel() {
+      if (this.isMailToURI) {
+        return null;
+      }
+      if (this.item.rel === false) {
+        return null;
+      }
+      if (this.item.rel) {
+        return this.item.rel;
+      }
+      return this.isBlankTarget ? "noopener noreferrer" : null;
```

Comment:
I learned recently that Safari and Chromium 88+ now default to behaving this way on `target="_blank"` which is great. Soon we will be able to drop this extra noise when IE and Legacy Edge support are out of favor.

source: https://web.dev/external-anchors-use-rel-noopener/